src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -60,86 +57,148 @@
                 } 
                 );
-            
-            } else if (browserSettingType.equals(CAPABILITIES)) {
+        }   
+        ); 
+        return browserSettings;
+    }
+    
+    protected static Map<String,Object> separateCapabilities(List<BrowserSetupElement> 
+        separateBrowserSettingsOsUnspecific) {
+        Map<String,Object> browserSettings = new HashMap();
+        separateBrowserSettingsOsUnspecific.forEach((bsetting) -> { 
+            bsetting.getCapabilities().forEach((capability) ->  { 
+                browserSettings.put(capability.getKey(), capability.getValue());
+            } 
+            );
+        });
+        return browserSettings;
+    }
+    
+    // just not deleted for your opinion. One method instead of 2 for options and capabilities    
+    //    protected static Map<String,Object> separateBrowserSettings(Map<String,Object> osUnspecificSettings,
+    //            List<BrowserSetupElement> separateBrowserSettingsOsspecific, String browserSettingType) {
+    //        Map<String,Object> browserSettings = new HashMap();
+    //        separateBrowserSettingsOsspecific.forEach((bsetting) -> { 
+    //            if (browserSettingType.equals(OPTIONS)) {
+    //                bsetting.getOptions().forEach((option) ->  {
+    //                    String key = option.getKey();
+    //                    if (osUnspecificSettings.containsKey(key)) {
+    //                        throw new RuntimeException("Duplicate entries existing in configuration file " 
+    //                            + FILE_NAME + ". Entries are: [" + key + " - " + option.getValue() + ", " 
+    //                            + key + " - " + osUnspecificSettings.get(key) + "]");
+    //                    } else {
+    //                        browserSettings.put(option.getKey(), option.getValue());
+    //                    }
+    //                } 
+    //                );
+    //        
+    //            } else if (browserSettingType.equals(CAPABILITIES)) {
+    //                bsetting.getCapabilities().forEach((capability) ->  { 
+    //                    String key = capability.getKey();
+    //                    if (osUnspecificSettings.containsKey(key)) {
+    //                        throw new RuntimeException("Duplicate entries existing in configuration file " 
+    //                            + FILE_NAME + ". Entries are: [" + key + " - " + capability.getValue() + ", " 
+    //                            + key + " - " + osUnspecificSettings.get(key) + "]");
+    //                    } else {
+    //                        browserSettings.put(capability.getKey(), capability.getValue());
+    //                    }
+    //                } 
+    //                );
+    //            }  
+    //    
+    //        }   
+    //        ); 
+    //        return browserSettings;
+    //    }
+    
+    protected static Map<String,Object> separateCapabilities(Map<String,Object> osUnspecificSettings,
```

Comment:
method name and parameter names are a bit misleading here, since this method is now very generic. How about copyCapabilitiesIntoMap(targetMap, browserSetupElements)